src/components/reusable/datePicker/datepicker.ts

```diff
@@ -79,135 +167,327 @@ export class DatePicker extends FormMixin(LitElement) {
+    }
+  }
+
+  private async reinitializeFlatpickr() {
+    this.flatpickrInstance?.destroy();
+    await this.initializeFlatpickr();
+  }
+
+  private async setupAnchor() {
+    if (this._inputEl) {
+      await this.initializeFlatpickr();
+    }
+  }
+
+  async initializeFlatpickr(): Promise<void> {
+    if (!this._inputEl) return;
+    if (this.flatpickrInstance) this.flatpickrInstance.destroy();
+
+    this.flatpickrInstance = await initializeSingleAnchorFlatpickr({
+      anchorEl: this._inputEl,
+      getFlatpickrOptions: this.getComponentFlatpickrOptions.bind(this),
+      setCalendarAttributes: (instance) =>
+        setCalendarAttributes(instance, 'Calendar'),
+      setInitialDates: this.setInitialDates.bind(this),
+      appendToBody: false,
+    });
+
+    this._validate(false, false);
+  }
+
+  setInitialDates(): void {
+    if (this.value && this.flatpickrInstance) {
+      this.flatpickrInstance.setDate(this.value, true);
+    }
+  }
 
-    // emit selected value
-    const event = new CustomEvent('on-input', {
-      detail: {
-        value: e.target.value,
-        origEvent: e,
-      },
+  async getComponentFlatpickrOptions(): Promise<Partial<BaseOptions>> {
+    return getFlatpickrOptions({
+      locale: this.locale,
+      dateFormat: this.dateFormat,
+      enableTime: this._enableTime,
+      twentyFourHourFormat: this.twentyFourHourFormat,
+      altFormat: this.altFormat,
+      startAnchorEl: this._inputEl!,
+      minDate: this.minDate,
+      maxDate: this.maxDate,
+      enable: this.enable,
+      disable: this.disable,
+      mode: this.mode,
+      closeOnSelect: !(this.mode === 'multiple' || this._enableTime),
+      onOpen: this.handleOpen.bind(this),
+      onClose: this.handleClose.bind(this),
+      onChange: this.handleDateChange.bind(this),
+      loadLocale,
     });
-    this.dispatchEvent(event);
   }
 
-  override updated(changedProps: PropertyValues) {
-    // preserve FormMixin updated function
-    this._onUpdated(changedProps);
+  handleOpen(): void {
+    if (!this._shouldFlatpickrOpen) {
+      this.flatpickrInstance?.close();
+      this._shouldFlatpickrOpen = true;
+    }
+  }
+
+  async handleClose(): Promise<void> {
+    this._hasInteracted = true;
+    this._validate(true, false);
+    await this.updateComplete;
+  }
+
+  async handleDateChange(
+    selectedDates: Date[],
+    dateStr: string
+  ): Promise<void> {
+    this._hasInteracted = true;
 
-    if (changedProps.has('value')) {
-      // set value on input element
-      this._inputEl.value = this.value;
+    if (this.mode === 'multiple') {
+      this.value = [...selectedDates];
+    } else {
+      this.value = dateStr;
     }
+
+    const formattedDates = Array.isArray(this.value)
+      ? this.value.map((date) => date.toISOString())
+      : this.value;
+
+    emitValue(this, 'on-change', {
+      dates: formattedDates,
+      dateString: (this._inputEl as HTMLInputElement)?.value || dateStr,
```

Comment:
We should mention the event output structure has changed.

Reply:
✅